src/job_cache/job_cache.cpp

```diff
@@ -167,28 +167,26 @@ wcl::optional<wcl::unique_fd> try_connect(std::string dir) {
   return wcl::make_some<wcl::unique_fd>(std::move(socket_fd));
 }
 
-Cache::Cache(std::string dir, uint64_t max, uint64_t low) {
-  mkdir_no_fail(dir.c_str());
-
-  // launch the daemon
-  if (daemonize(dir.c_str())) {
-    // We are the daemon, launch the cache
-
+// Launch the job cache daemon
+void Cache::launch_daemon() {
+  // We are the daemon, launch the cache
+  if (daemonize(cache_dir.c_str())) {
     std::string job_cache = wcl::make_canonical(find_execpath() + "/../bin/job-cache");
-    std::string low_str = std::to_string(low);
-    std::string max_str = std::to_string(max);
-    execl(job_cache.c_str(), "job-cached", dir.c_str(), low_str.c_str(), max_str.c_str(), nullptr);
+    std::string low_str = std::to_string(low_threshold);
+    std::string max_str = std::to_string(max_size);
+    execl(job_cache.c_str(), "job-cached", cache_dir.c_str(), low_str.c_str(), max_str.c_str(),
+          nullptr);
 
     wcl::log::fatal("exec(%s): %s", job_cache.c_str(), strerror(errno));
   }
+}
 
-  // connect to the daemon with backoff.
-  // TODO: Put this into a function so that we can call it again if
-  // the daemon fails unexpectedly.
+// Connect to the job cache daemon with backoff.
+void Cache::backoff_try_connect() {
   wcl::xoshiro_256 rng(wcl::xoshiro_256::get_rng_seed());
   useconds_t backoff = 1000;
   for (int i = 0; i < 10; i++) {
```

Comment:
While you're here, can you change this to 14 instead of 10? Nathan has reported this error-ing out and I think it was just because the machine was slow. Putting this up to 14 will allow for up to ~16 seconds of hang before killing itself while right now we wait ~1 second

Reply:
Actually I realized that we also want exponential back off on the read/write retries so I think we want this to be configurable so that on start up we give it the old college try but on retries we try to limit the wait to about 1 second.